tfs/src/main/java/hudson/plugins/tfs/JenkinsEventNotifier.java

```diff
@@ -0,0 +1,126 @@
+package hudson.plugins.tfs;
+
+import hudson.plugins.tfs.model.ConnectionParameters;
+import hudson.plugins.tfs.model.JobCompletionEventArgs;
+import hudson.plugins.tfs.util.TeamRestClient;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Formatter;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * This class sends Jenkins events to all "connected" team collections.
+ * Current list of events:
+ * - Job Completion event
+ */
+public class JenkinsEventNotifier {
+    protected final static Logger log = Logger.getLogger(JenkinsEventNotifier.class.getName());
+
+    /**
+     * Send the Job Completion event to connected TFS/VSTS servers
+     */
+    public static void sendJobCompletionEvent(final JSONObject payload) {
+        final List<TeamCollectionConfiguration> connectedCollections = TeamCollectionConfiguration.getConnectedCollections();
+        for (final TeamCollectionConfiguration c : connectedCollections) {
+            try {
+                // Check to see if there are any collections "connected" to this Jenkins server
+                final ConnectionParameters connectionParameters = c.getConnectionParameters();
+                if (connectionParameters == null || !connectionParameters.isSendJobCompletionEvents()) {
```

Comment:
good point. I will remove the null check and perhaps annotate the method with NotNull